utils/s2n_ensure.h

```diff
@@ -34,7 +34,14 @@
 
 #define __S2N_ENSURE_LIKELY( cond, action ) do {if ( s2n_unlikely( !(cond) ) ) { action; }} while (0)
 
-#define __S2N_ENSURE_CONDITION( cond, action ) __S2N_ENSURE_LIKELY( cond, action )
+#ifdef NDEBUG
+#define __S2N_ENSURE_DEBUG( cond, action ) do {} while (0)
+#else
+#define __S2N_ENSURE_DEBUG( cond, action ) __S2N_ENSURE_LIKELY((cond), action)
+#endif
+
+#define __S2N_ENSURE_PRECONDITION( result ) (s2n_likely(s2n_result_is_ok(result)) ? S2N_RESULT_OK : S2N_RESULT_ERROR)
+#define __S2N_ENSURE_POSTCONDITION( result ) (s2n_likely(s2n_result_is_ok(result)) ? S2N_RESULT_OK : S2N_RESULT_ERROR)
```

Comment:
Why are these needed?

Reply:
I had an `ifdef NDEBUG` that noop'd for the `POSTCONDITION` but it looks like it ended up being removed. @feliperodri was there a reason for that?

Reply:
@camshaft

Reply:
@camshaft I've added back the guard to noop for `POSTCONDITION`.